flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -2614,6 +2618,27 @@ private SqlNode registerFrom(
                         scopes.put(node, getSelectScope(call1.operand(0)));
                         return newNode;
                     }
+
+                    // Related to CALCITE-4077
+                    // ----- FLINK MODIFICATION BEGIN -----
+                    FlinkSqlCallBinding binding =
+                            new FlinkSqlCallBinding(this, getEmptyScope(), call1);
+                    if (op instanceof SqlVectorSearchTableFunction
+                            && binding.operand(0)
+                                    .isA(
+                                            new HashSet<>(
+                                                    Collections.singletonList(SqlKind.SELECT)))) {
+                        boolean queryColumnIsNotLiteral =
+                                binding.operand(2).getKind() != SqlKind.LITERAL;
+                        if (!queryColumnIsNotLiteral && !lateral) {
```

Comment:
Isn't `lateral` always needed? What's the syntax for literal?

Reply:
`LATERAL` is not always needed if the SqlCall doesn't contain correlation. For exmaple, users can use the following statement to search.
```
SELECT * FROM TABLE(VECTOR_SEARCH(TABLE VectorTable, DESCRIPTOR(`g`), ARRAY[1.5, 2.0], 10))
```

Here, the query input is `ARRAY[1.5, 2.0]`. 

CC https://github.com/apache/flink/blob/06c61b8c363ff4cb53e992428fa716a3c008bb6f/flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/stream/sql/VectorSearchTableFunctionTest.java#L95

Reply:
It's a valid sql. But we don't add literal related rule in physical phase, so planner can not translate the sql correctly.  But the exception indicates the planner can parse the statement correctly.

Reply:
Got it. I guess it's more clear to have something like https://github.com/apache/flink/pull/26553/files#diff-19970e8600e459e820e1310beed925a10450f695698257d85648e8114b5e5aaeR92 to indicate it's not invalid case.